docs/DesignDocs/AttestedTLSAPIs.md

```diff
@@ -0,0 +1,623 @@
+Attested TLS APIs
+==========================
+
+Motivation
+------------
+
+An important use case for an enclave is establishing
+an attested TLS channel.
+The attested TLS channel allows a remote party
+to authenticate the enclave identity during a TLS handshake,
+which brideges the data-in-use and data-in-transit protection.
+The Open Enclave SDK demonstrates this use case in the
+the [attested TLS samples](/samples/attested_tls) with the use
+of two APIs, `oe_get_attestation_certificate_with_evidence_v2` and
+`oe_verify_attestation_certificate_with_evidence_v2`.
+The `oe_get_attestation_certificate_with_evidence_v2` API
+allows a developer to obtain a self-signed certificate with
+the enclave evidence embedded, which then is used during
+the TLS handshake. More importantly, the API binds the
+public key of the certificate with the evidence such that
+the TLS channel based on the public key can be *attested*.
+To validate the certificate, the receiver uses the
+`oe_verify_attestation_certificate_with_evidence_v2` API,
+which assesses the trustworthiness of the evidence
+(based on the DCAP client). In addition, the API outputs a
+list of claims, allowing the developer to implement additional
+verification logic (e.g., authenticate the enclave identity such as `MRENCLAVE` in SGX).
+After a successful validation, the receiver continues the handshake
+and finally establishes an *attested* secure channel with the sender.
+
+While the two APIs are useful in the above end-to-end scenario,
+they have the following limitations:
+
+- Do not to support general use cases
+
+  A typical attestation procedure includes three roles: an attester, a relying party,
+  and a verifier. Two general models are as follows
+  (refer to [Remote Attestation Procedures Architecture](https://github.com/ietf-rats-wg/architecture/blob/master/draft-ietf-rats-architecture.md)
```

Comment:
A better URI is probably https://datatracker.ietf.org/doc/html/draft-ietf-rats-architecture